### PR TITLE
watcher: mock Base tests

### PR DESCRIPTION
**Pull request by panoel**

### CODE REVIEW

File: watcher/src/watchers/__tests__/BaseWatcher.test.ts [hunk 1]
```diff
@@ -1,15 +1,73 @@
...
+        .replace('https://', '')}`;
+      if (existsSync(mockDataPath)) {
+        return { data: JSON.parse(readFileSync(mockDataPath, 'utf8')) };
+      }
+      return originalAxios.get(url, config);
+    }),
+    post: jest.fn(async (url: string, config?: any) => {
+      let mockDataPath = `${__dirname}/mock/${url.replace('http://', '').replace('https://', '')}`;
+      // console.log('post config', stringify(config));
+      if (config) {
```
[the remark this answers is not in the document]
Comment (evan-gray) [reply watcher/src/watchers/__tests__/BaseWatcher.test.ts:22]: only downside is having to figure it out with `console.log(data, dataHash)` the first time around and it being a little less obvious what is being pulled.

File: watcher/src/watchers/__tests__/BaseWatcher.test.ts [hunk 2]
```diff
@@ -1,15 +1,73 @@
 import { expect, jest, test } from '@jest/globals';
 import { INITIAL_DEPLOYMENT_BLOCK_BY_NETWORK_AND_CHAIN } from '@wormhole-foundation/wormhole-monitor-common';
 import { EVMWatcher } from '../EVMWatcher';
+import { existsSync, readFileSync } from 'fs';
 
 jest.setTimeout(60000);
+jest.mock('axios', () => {
```
Comment (evan-gray) [inline watcher/src/watchers/__tests__/BaseWatcher.test.ts:7]: probably worth moving this to `setup-jest.ts` so it's only in the one place.

**Review by evan-gray — CHANGES_REQUESTED** [review]
Might be worth trying a slightly more explicit, yet slightly more tedious approach. Also a couple quick signature fixes.

File: watcher/src/watchers/__tests__/BaseWatcher.test.ts [hunk 3]
```diff
@@ -1,15 +1,73 @@
...
+          mockDataPath = `${mockDataPath}/default`;
+        }
+      }
+      // console.log('post mockDataPath', mockDataPath);
+      if (existsSync(mockDataPath)) {
+        return JSON.parse(readFileSync(mockDataPath, 'utf8'));
+      }
+      // console.log('axios.post', url);
+      const retval = await originalAxios.post(url, config);
+      // console.log('post retval', stringify(retval));
```
[the remark this answers is not in the document]
Comment (panoel) [reply watcher/src/watchers/__tests__/BaseWatcher.test.ts:43]: Fixed

File: watcher/src/watchers/__tests__/BaseWatcher.test.ts [hunk 4]
```diff
@@ -1,15 +1,73 @@
...
+    get: jest.fn(async (url: string, config?: any) => {
+      const mockDataPath = `${__dirname}/mock/${url
+        .replace('http://', '')
+        .replace('https://', '')}`;
+      if (existsSync(mockDataPath)) {
+        return { data: JSON.parse(readFileSync(mockDataPath, 'utf8')) };
+      }
+      return originalAxios.get(url, config);
+    }),
+    post: jest.fn(async (url: string, config?: any) => {
```
[the remark this answers is not in the document]
Comment (panoel) [reply watcher/src/watchers/__tests__/BaseWatcher.test.ts:19]: Fixed

File: watcher/src/watchers/__tests__/BaseWatcher.test.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (panoel) [reply watcher/src/watchers/__tests__/BaseWatcher.test.ts:22]: Won't this, also, apply to the getting latest block?

[the remark this answers is not in the document]
Comment (panoel) [reply watcher/src/watchers/__tests__/BaseWatcher.test.ts:22]: Changed

File: watcher/src/watchers/__tests__/BaseWatcher.test.ts (hunk 2 above)
Comment (panoel) [reply watcher/src/watchers/__tests__/BaseWatcher.test.ts:7]: Moved

File: watcher/src/watchers/__tests__/BaseWatcher.test.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (evan-gray) [reply watcher/src/watchers/__tests__/BaseWatcher.test.ts]: only what exists in the folder :+1:

File: watcher/setup-jest.ts [hunk 5]
```diff
@@ -1,4 +1,38 @@
...
+    post: jest.fn(async (url: string, data?: any, config?: any) => {
+      const dataHash = Buffer.from(
+        await crypto.subtle.digest('SHA-256', Buffer.from(JSON.stringify(data)))
+      ).toString('hex');
+      const mockDataPath = `${__dirname}/mock/${url
+        .replace('http://', '')
+        .replace('https://', '')}/${dataHash}`;
+      // console.log('post mockDataPath', mockDataPath);
+      if (existsSync(mockDataPath)) {
+        return JSON.parse(readFileSync(mockDataPath, 'utf8'));
```
Comment (evan-gray) [inline watcher/setup-jest.ts:30]: can this store and return just the data or does that not work?
e.g. the `return { data: JSON.parse(readFileSync(mockDataPath, 'utf8')) };` like above

**Review by evan-gray — CHANGES_REQUESTED** [review]
Nice cleanup. Just one more nit, probably don't need to store the whole axios object.

File: watcher/setup-jest.ts (hunk 5 above)
Comment (panoel) [reply watcher/setup-jest.ts:30]: Fixed.

**Review by evan-gray — APPROVED** [review]
Thanks!

**Pull request merged**